### PR TITLE
Fix F10 menu crash caused under specific circumstances

### PR DESCRIPTION
**Pull request by ricmzn**
Message from commit:

missionCommands.addMenu accepts multiple user arguments to the handler
function, however, it seems that behavior is not documented, and the
extra parameters can cause memory corruption under certain circumstances
when running queued commands on units that have been removed by the slot
kicker. By reducing the user arguments to one, the functions behave
correctly and DCS doesn't crash anymore when calling them. The
theater object used to be passed in the commands, but now it is injected
directly into the playerRequest method via the singleton call (as it was
done in groupmenu.lua before).

### CODE REVIEW

**Comment by ricmzn** [comment]
I believe I fixed it by giving Theater instances a `scratchpad` table, though what is it used for? Can cmds.lua:81 be removed? I don't see the scratchpad table in Theater referenced anywhere else, only the (different) one in player assets.

**Comment by ricmzn** [comment]
Nevermind the above fix, scratchpad functionality needs to be fixed in detail first as it's referencing a different table in the scratchpad class itself.

File: src/dct/Theater.lua [hunk 1]
```diff
@@ -99,6 +99,7 @@ function Theater:__init()
 	self._systems  = {}
 	self.startdate = os.date("!*t")
 	self.namecntr  = 1000
+	self.scratchpad = {}
```
Comment (jtoppins) [inline src/dct/Theater.lua:102]: I don't see why you need this.

File: tests/test-ui-cmds2.lua [hunk 2]
```diff
@@ -61,7 +61,7 @@ local function main()
 	for _, v in ipairs(testcmds) do
 		trigger.action.setassert(v.assert)
 		trigger.action.setmsgbuffer(v.expected)
-		theater:playerRequest(v.data)
+		theater.playerRequest(v.data)
```
Comment (jtoppins) [inline tests/test-ui-cmds2.lua:64]: You want to use `dct.Theater.playerRequest`.

File: src/dct/Theater.lua (hunk 1 above)
Comment (jtoppins) [reply src/dct/Theater.lua:102]: What was the error message you were trying to fix?

Comment (jtoppins) [reply src/dct/Theater.lua:102]: This should be fixed with PR #199

**Pull request closed without merging**